WIP MTV 2.5.0 release notes (#414)
kubev2v/forklift-documentation: documentation/modules/rn-2.5.adoc

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-2.5_{context}"]
= {project-full} 2.5

> Migrate virtual machines (VMs) from VMware vSphere or {rhv-full} or {osp} to {virt} with {the-lc} {project-first}.

Author: I will add links when the documents are available, but any text you want to include, please feel free to add

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-25_{context}"]

> == Technical changes

Author: Yes sure, please can I ask for more details and anything else you want to include?
I am happy to go into as much detail as you want. I feel we need to improve the quality of the release notes, so happy to push change here

> == Technical changes

Reviewer: added a coment in place

This release has the following technical changes:

[id="new-features-and-enhancements-25_{context}"]
== New features and enhancements

This release has the following features and improvements:

[id="known-issues-25_{context}"]
== Known issues

This release has the following known issues:

////

> Adding 2.4 known issues as a start. i am not sure there is much point including the other sections

Reviewer 2: right

VM with multiple disks that was migrated might not be able to boot on the target OCP cluster. Workaround: Set the boot order appropriately to boot from the bootable disk. link:https://issues.redhat.com/browse/MTV-433[(MTV-433)]

.Non-supported guest operating systems in warm migrations

Warm migrations and migrations to remote OCP clusters from vSphere do not support all types of guest operating systems that are supported in cold migrations to the local OCP cluster. It is a consequence of using RHEL 8 in the former case and RHEL 9 in the latter case. +
See link:https://access.redhat.com/articles/1351473[Converting virtual machines from other hypervisors to KVM with virt-v2v in RHEL 7, RHEL 8, and RHEL 9] for the list of supported guest operating systems.

.VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down

When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in OpenShift Virtualization. link:https://issues.redhat.com/browse/MTV-491[(MTV-491)]

.Upgrade from 2.4.0 fails

> When upgrading from MTV 2.4.0 to a later version, the operation fails with an error that says the field 'spec.selector' of deployment `forklift-controller` is immutable. Workaround: remove the custom resource `forklift-controller` of type `ForkliftController` from the installed namespace, and recreate it. The user needs to refresh the OCP Console once the `forklift-console-plugin` pod runs to load the upgraded {project-short} web console. link:https://issues.redhat.com/browse/MTV-518[(MTV-518)]

Author: I am adding a section called upgrade notes, if that is ok (see line 135)

> For a complete list of all known issues in this release, see the list of link:https://issues.redhat.com/browse/MTV-562?filter=12419159[Known Issues] in Jira. 

Reviewer 2: nice, I need to go over that list - already saw a duplicate issue there

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-2.5_{context}"]
= {project-full} 2.5

Migrate virtual machines (VMs) from VMware vSphere or {rhv-full} or {osp} to {virt} with {the-lc} {project-first}.

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-25_{context}"]
== Technical changes

This release has the following technical changes:

[id="new-features-and-enhancements-25_{context}"]
== New features and enhancements

> This release has the following features and improvements:

Reviewer 3: @ahadas  Did you mean,  "It is strongly recommended to create a VDDK init image to accelerate migrations. For more information, see xref:../master.adoc#creating-vddk-image_mtv[Creating a VDDK image]." ?

> This release has the following features and improvements:

Author: @RichardHoch - thanks will change

.Migration using OVA files that are compatible with VMware vSphere

> // i need to wait for this ticket to be merged to add a link

Author: Sorry, I meant link to the documentation. Once Richard begins merging I can `xref` to the doc sections.

In {project-short} {project-version}, you can migrate using Open Virtual Application (OVA) files that were created by or are compatible with VMware vSphere as source providers. Although OVA files created by vSphere are supported, it may also work with OVAs that are created by other systems and compatible with vSphere.

.Migrating VMs between OCP clusters

> // i need to wait for this ticket to be merged to add a link

Author: Sorry, I meant link to the documentation. Once Richard begins merging I can `xref` to the doc sections.

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-2.5_{context}"]
= {project-full} 2.5

You can use {the-lc} {project-first} to migrate virtual machines from the following source providers to {virt} destination providers:

* VMware vSphere
* {rhv-full} ({rhv-short})
* {osp}
* Open Virtual Appliances (OVAs) that were created by VMware vSphere
* Remote {virt} clusters

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-25_{context}"]
== Technical changes

This release has the following technical changes:

.Migration using OpenStack

In this version, migration using OpenStack source providers graduated from a Technology Preview feature to a fully supported feature.

.Disabling FIPS

EMS enforcement is disabled for migrations with VMware vSphere source providers in order to enable migrations from versions of vSphere that are supported by {project-full} but do not comply with the 2023 FIPS requirements.

.Integration of the create/update provider dialog

>  The forms of create and update provider are replaced with native pages within the {ocp-name}(OCP) console.

Reviewer: ```suggestion
 The forms of create and update provider are replaced with native pages within the {ocp-name}(OCP) console.
```

the end user does not "see" the change to native pages, it's an implementation detail,
but they do see a design change
we can say 
the forms of create and update provider where re designed to align with the ocp console design and added settings of the new Openstack authentication,
make VDDK optional 
allow use of system CA  
add OVA settings

Reviewer 2: @yaacov ok, maybe I mixed things here - was it the create-plan dialog that wasn't integrated well into the console and triggered a separate window?

yeah, if the create/update provider dialog was already integrated into the console and "just" reshaped to align with the OCP console standard then this need to be rephrased

about the list of addition at the end - it's a release note, we don't have to list all those things as they are mentioned in separate items

Reviewer: in this release we only updated the providers dialog, it was using none ocp api and ui components, so updating to use ocp api and components changed the way it looks and feel to align with other parts of the console, and some data inconsistencies like values not getting updated was fixed

Reviewer 2: ok, so let's try to summarize it in a release-note way:
```suggestion
Reworked the forms of create and update providers so that they will align with the typical look and feel within the OCP console and ensure the latest data is presented in these dialogs.
```

Reviewer 2: or actually, we have [a bug on the second part](https://issues.redhat.com/browse/MTV-603) so we can go with
```suggestion
 Reworked the forms of create and update providers so that they will align with the typical look and feel within the OCP console.
```

[id="resolved-issues-25_{context}"]
== Resolved issues

// are there any resolved issues you want to highlight in this release?

.Snapshots that are created during the migration in OpenStack are not deleted

In previous releases of {project-short}, the Migration Controller service did not delete snapshots that were created during the migration for source virtual machines in OpenStack automatically.

This issue is resolved in {project-short} {project-version}, all the snapshots created during the migration are removed after the migration has been completed successfully. link:https://issues.redhat.com/browse/MTV-620[MTV-620]

.{rhv-short} snapshots are not deleted after a successful migration

In previous releases of {project-short}, the Migration Controller service did not delete snapshots automatically after a successful warm migration of a VM from {rhv-short}.

This issue is resolved in {project-short} {project-version}, the snapshots generated during migration are removed after a successful migration, and the original snapshots are not removed after a successful migration. link:https://issues.redhat.com/browse/MTV-349[(MTV-349)]

> .Migration fails during precopy/cutover while a snapshot operation is executed on the source VM

Reviewer 2: ok, this item should change:
1. we need to keep a known issue that is correlated with https://issues.redhat.com/browse/MTV-456 and says that warm migration fails when a snapshot is created on the source VM not by MTV (snapshot that is created in RHV by the user / other tool)
2. this item should include two parts: (a) Cutover can not be triggered when there's an ongoing precopy operation (https://issues.redhat.com/browse/MTV-686); and if https://issues.redhat.com/browse/MTV-687 gets verified for 2.5.0 then (b) Warm migration doesn't fail when an operation that locks the VM is executed in RHV.

about (2), we can split this item to two items or combine them under a single one, it's title can be something like "Improve warm migrations from RHV" or maybe more specifically "Resolved failures of warm migrations from RHV due to timing issues"

Author: [MTV-686](https://issues.redhat.com/browse/MTV-686) is on verified - add as **Resolved Issue**?

[MTV-687](https://issues.redhat.com/browse/MTV-687) is on verified - add as **Resolved Issue**?

Author: Added [MTV-686](https://issues.redhat.com/browse/MTV-686) as a resolved issue as the status is VERIFIED

Will give [MTV-687](https://issues.redhat.com/browse/MTV-687) a day or two to see if moves to verified

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-2.5_{context}"]
= {project-full} 2.5

You can use {the-lc} {project-first} to migrate virtual machines from the following source providers to {virt} destination providers:

* VMware vSphere
* {rhv-full} ({rhv-short})
* {osp}
* Open Virtual Appliances (OVAs) that were created by VMware vSphere
* Remote {virt} clusters

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-25_{context}"]
== Technical changes

This release has the following technical changes:

> .Migration using OpenStack

Reviewer 2: and there's something confusing about this title since migrations from openstack were introduced in 2.4 - maybe "Migration from OpenStack becomes fully-supported"?

> .Migration using OpenStack

Author: changed

In this version, migration using OpenStack source providers graduated from a Technology Preview feature to a fully supported feature.

.Disabling FIPS

EMS enforcement is disabled for migrations with VMware vSphere source providers in order to enable migrations from versions of vSphere that are supported by {project-full} but do not comply with the 2023 FIPS requirements.

.Integration of the create/update provider dialog

Reworked the forms of create and update providers so that they will align with the typical look and feel within the OCP console and ensure the up-to-date data is displayed.

.Standalone UI

The old UI of MTV 2.3 cannot be enabled by setting `feature_ui: true` in ForkliftController anymore.

[id="new-features-and-enhancements-25_{context}"]
== New features and enhancements

This release has the following features and improvements:

> .Migration using OVA files that are compatible with VMware vSphere

Reviewer 2: ack, I'll comment per-item

// i need to wait for this ticket to be merged to add a link
In {project-short} {project-version}, you can migrate using Open Virtual Application (OVA) files that were created by VMware vSphere as source providers. It may also work with OVAs that are created by other systems and compatible with vSphere.

.Migrating VMs between OCP clusters

// i need to wait for this ticket to be merged to add a link
// apologies, I meant link to output of the doc ticket rather than https://issues.redhat.com/browse/MTV-336[Import VMware-compatible OVAs]
// But let me add this as a placeholder 

> You can import Virtual Machine (VMs) without having access from {ocp-name} to the source system the VMs were running on. The VMs are exported to a place that is accessible to {ocp-name}, optionally accessible to the source system at the same time.

Author: changed

.Migration of VMs with direct LUNs from RHV 

During the migration from RHV, direct LUNs are detached from the source virtual machines and attached to the target virtual machines. Note that this mechanism does not work yet for Fibre Channel. link:https://issues.redhat.com/browse/MTV-329[MTV-329]

> .Additional authentication methods for OpenStack

Reviewer 4: The additional authentication methods are not working properly now, should be add the bug https://issues.redhat.com/browse/MTV-701 and tell user the workaround?

Reviewer 2: actually, maybe it better be documented as a known issue and add a link to the CLI section there

Author: I have added a workaround. We can move in to the known issues next week

Reviewer 2: ack

In addition to the standard password authentication, the following authentication methods are supported: Token authentication and Application credential authentication. link:https://issues.redhat.com/browse/MTV-539[MTV-539]

.Validation rules for OpenStack

The validation service includes default validation rules for virtual machines from OpenStack. link:https://issues.redhat.com/browse/MTV-508[MTV-508]

> .Import image-based virtual machines from OpenStack Migration

Author: Done

See link:https://access.redhat.com/articles/1351473[Converting virtual machines from other hypervisors to KVM with virt-v2v in RHEL 7, RHEL 8, and RHEL 9] for the list of supported guest operating systems.

.VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down

When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in {ocp-name} Virtualization. link:https://issues.redhat.com/browse/MTV-491[(MTV-491)]

.Import OVA: ConnectionTestFailed message appears when adding OVA provider

When adding an OVA provider, the error message `ConnectionTestFailed` appears, although the provider is created successfully. After a few seconds, this message disappears. link:https://issues.redhat.com/browse/MTV-671[MTV-671]

.OVA migration with multiple disks is not functioning

OVA migration with multiple disks is not functioning as expected. When trying to migrate an OVA VM, with more than one disk, the migration becomes stuck at the `allocate disks` phase. link:https://issues.redhat.com/browse/MTV-676[MTV-676]

> 

Author: @ahadas && @bkhizgiy && @qiyuann - adding the final two Known Issues 
Please can you add your comments, so we can have this ready for Thursday

Reviewer 5: Looking good overall, regarding the first issue I think we need rephrase it a bit, because the issue can also indicate on a real connection problem (not part of the current bug) and not disappear in that case, we should write something like `"When adding an OVA provider, the error message ConnectionTestFailed may instantly appear, although the provider was created successfully. If after a few minutes the message isn't disappearing and the provider status turns to ready condition, the ova server pod creation really failed."`

Author: Change made

For a complete list of all known issues in this release, see the list of link:https://issues.redhat.com/browse/MTV-562?filter=12419159[Known Issues] in Jira.

[id="resolved-issues-25_{context}"]
== Resolved issues

// are there any resolved issues you want to highlight in this release?

> .Ensure up-to-date data is displayed in the create and update provider form

Reviewer 2: `forms`, no?

Reviewer 2: ```suggestion
.Ensure up-to-date data is displayed in the create and update provider forms
```

In previous releases of {project-short}, the create and update provider forms could have presented stale data.

This issue is resolved in {project-short} {project-version}, the new forms of create and update provider display up-to-date properties of the provider. link:https://issues.redhat.com/browse/MTV-603[MTV-603]

.Snapshots that are created during the migration in OpenStack are not deleted

> In previous releases of {project-short}, the Migration Controller service did not delete snapshots that were created during the migration for source virtual machines in OpenStack automatically.

Reviewer 2: ```suggestion
In previous releases of {project-short}, the Migration Controller service did not delete snapshots that were created during the migration of source virtual machines in OpenStack automatically.
```
or maybe we should rather say ".. that were created automatically during the migration from OpenStack" ?

> This issue is resolved in {project-short} {project-version}, all the snapshots created during the migration are removed after the migration has been completed successfully. link:https://issues.redhat.com/browse/MTV-620[MTV-620]

Reviewer 2: ```suggestion
This issue is resolved in {project-short} {project-version}, all the snapshots created during the migration are removed after the migration has been completed. link:https://issues.redhat.com/browse/MTV-620[MTV-620]
```
also when the migration ends with failure

.{rhv-short} snapshots are not deleted after a successful migration

In previous releases of {project-short}, the Migration Controller service did not delete snapshots automatically after a successful warm migration of a VM from {rhv-short}.

This issue is resolved in {project-short} {project-version}, the snapshots generated during migration are removed after a successful migration, and the original snapshots are not removed after a successful migration. link:https://issues.redhat.com/browse/MTV-349[(MTV-349)]

> .Migration fails during precopy/cutover while a snapshot operation is executed on the source VM

Reviewer 2: this one can be dropped in favor of the next two resolve issues

This issue is resolved in {project-short} {project-version}, {project-short} waits for the snapshot operation to finish in order to run the next cutover. link:https://issues.redhat.com/browse/MTV-456[(MTV-456)]

.Warm migration fails when cutover conflicts with precopy

In previous releases of {project-short}, the cutover operation failed when it was triggered while precopy was being performed. The VM was locked in {rhv-short} and therefore the `ovirt-engine` rejected the snapshot creation, or disk transfer, operation.

This issue is resolved in {project-short} {project-version}, the cutover operation is triggered, but it is not performed that time because the VM is locked. Once the precopy operation completes, the cutover operation is triggered. link:https://issues.redhat.com/browse/MTV-686[MTV-686]

.Warm migration fails when VM is locked

In previous release of {project-short}, when triggering warm migration while there was an ongoing operation in {rhv-short} that locked the VM. The migration failed because the snapshot creation could not be triggered. 

> This issue is resolved in {project-short} {project-version}, warm migration does not fail when an operation that locks the VM is performed in {rhv-short}. The migration does not fail, but starts when the VM is unlocked. link:https://issues.redhat.com/browse/MTV-686[MTV-686] 

Reviewer 2: ```suggestion
This issue is resolved in {project-short} {project-version}, warm migration does not fail when an operation that locks the VM is performed in {rhv-short}. The migration does not fail, but starts when the VM is unlocked. link:https://issues.redhat.com/browse/MTV-687[MTV-687] 
```

.PVC deletion hangs after archiving and deleting migration plan

In previous releases of {project-short}, when a migration failed, its PVCs and PVs were not deleted as expected when its migration plan was archived and deleted.

This issue is resolved in {project-short} {project-version}, PVCs are deleted when archiving and deleting migration plan.link:https://issues.redhat.com/browse/MTV-493[(MTV-493)]

.VM with multiple disks may boot from non-bootable disk after migration

In previous releases of {project-short}, VM with multiple disks that were migrated might not have been able to boot on the target OCP cluster.

This issue is resolved in {project-short} {project-version}, VM with multiple disks that are migrated are able to boot on the target OCP cluster. link:https://issues.redhat.com/browse/MTV-433[(MTV-433)]

For a complete list of all resolved issues in this release, see the list of link:https://issues.redhat.com/browse/MTV-433?filter=12419160[Resolved Issues] in Jira.
This release has the following resolved issues:

[id="upgrade-notes-25_{context}"]
== Upgrade notes

// Is there anything else you want me to add?

> It is recommended to upgrade from {project-short} 2.4.2 to {project-short} {project-version}.

Reviewer 2: this week we identified an issue with upgrades from 2.4.1 or 2.4.2 as well.. we need to note here that when upgrading from 2.4 (2.4.z), after the 2.5 operator starts, the mutatingwebhookconfiguration and validatingwebhookconfiguration named forklift-api should be removed:

oc -n openshift-mtv delete mutatingwebhookconfiguration forklift-api
oc -n openshift-mtv delete validatingwebhookconfiguration forklift-api

 and restart the operator

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-2.5_{context}"]
= {project-full} 2.5

You can use {the-lc} {project-first} to migrate virtual machines from the following source providers to {virt} destination providers:

* VMware vSphere
* {rhv-full} ({rhv-short})
* {osp}
* Open Virtual Appliances (OVAs) that were created by VMware vSphere
* Remote {virt} clusters

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-25_{context}"]
== Technical changes

This release has the following technical changes:

> .Migration using OpenStack moves to being a fully supported feature

Reviewer 2: ```suggestion
.Migration from OpenStack moves to being a fully supported feature
```
(I'm not sure if you've noticed it in my previous comment, if you did and it's better that way - that's fine)

In this version, migration using OpenStack source providers graduated from a Technology Preview feature to a fully supported feature.

.Disabling FIPS

EMS enforcement is disabled for migrations with VMware vSphere source providers in order to enable migrations from versions of vSphere that are supported by {project-full} but do not comply with the 2023 FIPS requirements.

.Integration of the create/update provider dialog

Reworked the forms of create and update providers so that they will align with the typical look and feel within the OCP console and ensure the up-to-date data is displayed.

.Standalone UI

The old UI of MTV 2.3 cannot be enabled by setting `feature_ui: true` in ForkliftController anymore.

[id="new-features-and-enhancements-25_{context}"]
== New features and enhancements

This release has the following features and improvements:

> .Migration using OVA files that are compatible with VMware vSphere

Reviewer 2: jira epic for this one: https://issues.redhat.com/browse/MTV-336

// i need to wait for this ticket to be merged to add a link
In {project-short} {project-version}, you can migrate using Open Virtual Application (OVA) files that were created by VMware vSphere as source providers. It may also work with OVAs that are created by other systems and compatible with vSphere.

> .Migrating VMs between OCP clusters

Reviewer 2: jira epic for this one: https://issues.redhat.com/browse/MTV-571

// i need to wait for this ticket to be merged to add a link
// apologies, I meant link to output of the doc ticket rather than https://issues.redhat.com/browse/MTV-336[Import VMware-compatible OVAs]
// But let me add this as a placeholder 

> In {project-short} {project-version}, you can also use Red Hat OpenShift Virtualization provider as a source provider. You can migrate VMs from the cluster that MTV is deployed on to another cluster, or from a remote cluster to the cluster that {project-short} is deployed on. link:https://issues.redhat.com/browse/MTV-571[MTV-571]

Reviewer 2: ```suggestion
In {project-short} {project-version}, you can also use Red Hat {virt} provider as a source provider. You can migrate VMs from the cluster that MTV is deployed on to another cluster, or from a remote cluster to the cluster that {project-short} is deployed on. link:https://issues.redhat.com/browse/MTV-571[MTV-571]
```
